activesupport/lib/active_support/railtie.rb

```diff
@@ -87,6 +87,10 @@ class Railtie < Rails::Railtie # :nodoc:
         if app.config.active_support.use_sha1_digests
           ActiveSupport::Digest.hash_digest_class = ::Digest::SHA1
         end
+
+        if klass = app.config.active_support.use_digest_class
```

Comment:
Should this be an `elsif` instead? It could be confusing if you set `use_sha1_digests` to true, only to find out it gets overridden by `use_digest_class` to something else.

Or maybe it would be better to deprecate `use_sha1_digests` instead and only use `use_digest_class`. In that case I'd prefer the name `hash_digest_class`.

Reply:
> Should this be an `elsif` instead? It could be confusing if you set `use_sha1_digests` to true, only to find out it gets overridden by `use_digest_class` to something else.

The same argument can be made that if it's an `elsif` and you end up with both, the second option doesn't seem to work. 

In the end I don't have a strong opinion on this though, ideally Rails has some example behavior that was defined before that we can follow here.

> Or maybe it would be better to deprecate `use_sha1_digests` instead and only use `use_digest_class`. In that case I'd prefer the name `hash_digest_class`.

That certainly sounds like a good option to me. A boolean flag for something where multiple algorithms can be used, seems like not the right way to set it and being able to configure the actual hash would make sense to me. I have no strong preferences on the name here, so if there's consensus on a better name, I'm all for changing it.

Reply:
+1 for deprecating the boolean `use_sha1_digests` and introducing `hash_digest_class`.

Reply:
@p8 @jeremy I've updated the PR here with a deprecation and using `hash_digest_class` as the name instead. Is this looking any good?